tidy3d/components/monitor.py

```diff
@@ -158,6 +158,32 @@ class AbstractFieldMonitor(Monitor, ABC):
         description="Collection of field components to store in the monitor.",
     )
 
+    interval_space: Tuple[
+        pydantic.PositiveInt, pydantic.PositiveInt, pydantic.PositiveInt
+    ] = pydantic.Field(
+        (1, 1, 1),
+        title="Spatial interval",
+        description="Number of grid step intervals between monitor recordings. If equal to `1`, "
+        "there will be no downsampling. If greater than 1, fields will be downsampled "
+        "and automatically colocated.",
+    )
+
+    colocate_space: bool = pydantic.Field(
+        None,
+        title="Colocate fields",
+        description="Toggle whether fields should be colocated to grid cell centers. Default: "
+        "`False` if `interval_space` is 1 in each direction, `True` if `interval_space` is "
+        "greater than one in any direction.",
+    )
+
+    @pydantic.validator("colocate_space", always=True)
+    def set_default_colocate_space(cls, val, values):
```

Comment:
is there a situation where we want to restrict the value of `colocate_space`?  For example, if `sum(interval_space) > 3` do we want to enforce that `colocate_space=True`? or not?

Reply:
What you describe is currently the default behaviour because it's likely to be the most frequent use case, but the solver permits colocating regardless of downsampling, so I figured we might as well allow each possibility?

Reply:
ok sounds good